### ISSUE TITLE
Officially deprecate this version of the plugin?

### ISSUE THREAD

**Issue by motiz88** [babel-plugins/babel-plugin-object-assign#1]
`babel-plugin-object-assign` is published [on npm](https://www.npmjs.com/package/babel-plugin-object-assign) with a link back here. However, the current plugin for Babel v6 is named [`babel-plugin-transform-object-assign`](https://babeljs.io/docs/plugins/transform-object-assign/) (note the added `transform`).

Accidentally installing `babel-plugin-object-assign` with Babel v6 results in a failure at build time (`TypeError: Plugin is not a function`) due to the differing plugin APIs.

IMHO this should fail (or at least warn) earlier and more clearly, using e.g. a versioned `npm` (peer) dependency on Babel v5, an `npm` deprecation notice in favor of the new package, etc.

**Comment by thejameskyle** [comment]
Yes, we should deprecate it on npm. I don't have access to the package on npm or I would.

**Comment by marksyzm** [comment]
Yeah, please do :) Had to get this far to notice

**Comment by jasonmendes** [comment]
I just came here with the same error as @motiz88 -- glad at least to have figured out the issue.

**Comment by diffractometer** [comment]
I also came here with the same error.

**Comment by hzoo** [comment]
Do we actually want to unpublish? Its just that we don't want a user to use this for babel 6 since people could be using this for babel 5.

cc @jedwatson since npm

**Comment by marksyzm** [comment]
Good point. But is there a way to give a warning to Babel users, instead?

**Comment by hzoo** [comment]
Yeah I'l try to work on that soon (a babel change)

**Comment by hzoo** [comment]
https://github.com/babel/babel/pull/3362

**Comment by RReverser** [comment]
@hzoo You could use `peerDependency` for that as well. With npm@3 new behavior, it will just throw an error if installed Babel version doesn't match expected from `peerDependency`. In npm@2 it will also throw an error if Babel 6 is found, otherwise it will attempt to install Babel 5 on its own - which is still much better than what happens currently.

**Comment by RReverser** [comment]
TL;DR - just add peerDependency for babel@^5 and it will always refuse to install on top of Babel 6.

**Comment by hzoo** [comment]
I was goina make it generic unless we want to add that for all the babel 5 plugins
